Error when deploying for the first time with certain stage names

Hello, when deploying for the first time in a directory it asks for a stage name:
`Look like you’re running sst for the first time in this directory. Please enter a stage name you’d like to use locally. Or hit enter to use the one based on your AWS credentials`

If you enter a stage name with spaces it throws this error:

```
Preparing your SST app

=======================
 Deploying debug stack
=======================

node:internal/errors:465
    ErrorCaptureStackTrace(err);
    ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'C:\Users\goto\Documents\Proyectos\Proyecto\sst\examples\react-app\node_modules\@serverless-stack\cli\assets\debug-stack\us-east-1\.build\lib\index.js' imported from C:\Users\goto\Documents\Proyectos\Proyecto\sst\examples\react-app\node_modules\@serverless-stack\cli\assets\debug-stack\bin\index.mjs
    at new NodeError (node:internal/errors:372:5)
    at finalizeResolution (node:internal/modules/esm/resolve:437:11)
    at moduleResolve (node:internal/modules/esm/resolve:1009:10)
    at defaultResolve (node:internal/modules/esm/resolve:1218:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:580:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:294:18)
    at ESMLoader.import (node:internal/modules/esm/loader:380:22)
    at importModuleDynamically (node:internal/modules/esm/translators:106:35)
    at importModuleDynamicallyCallback (node:internal/process/esm_loader:35:14)
    at file:///C:/Users/goto/Documents/Proyectos/Proyecto/sst/examples/react-app/node_modules/@serverless-stack/cli/assets/debug-stack/bin/index.mjs:21:51 {
  code: 'ERR_MODULE_NOT_FOUND'
}
```

if I set a stage name a word with no spaces this does not happen.
It would be nice if it showed a nicer error message or better yet, replace spaces with `-`, I spent like 30 minutes figuring what it was.

Thanks

--- a/.changeset/sixty-timers-walk.md
+++ b/.changeset/sixty-timers-walk.md
@@ -1,0 +1,6 @@
+---
+"@serverless-stack/cli": patch
+"@serverless-stack/core": patch
+---
+
+Cli: check for invalid stages names

--- a/packages/cli/bin/scripts.mjs
+++ b/packages/cli/bin/scripts.mjs
@@ -261,21 +261,35 @@ async function getStage(argv, config) {
 
   if (process.env.__TEST__ === "true") return DEFAULT_STAGE;
 
+  // Generate a suggested stage name as the default
   const suggested = await State.suggestStage();
+  const question = `Please enter a stage name you’d like to use locally. Or hit enter to use the one based on your AWS credentials (${suggested}): `;
+
+  // Prompt to enter a stage name
+  const input = await questionSync(`Look like you’re running sst for the first time in this directory. ${question}`);
+  let final = input.trim() || suggested;
+
+  // Re-prompt if stage name is invalid
+  while(!State.validateStage(final)) {
+    logger.error(chalk.red("Stage name must start with a letter, and contain only letters, numbers, and hyphens."));
+    const input = await questionSync(`\n${question}`);
+    final = input.trim() || suggested;
+  }
+
+  State.setStage(paths.appPath, final);
+  return final;
+}
+
+function questionSync(question) {
   const rl = readline.createInterface({
     input: process.stdin,
     output: process.stdout,
   });
   return new Promise((resolve) => {
-    rl.question(
-      `Look like you’re running sst for the first time in this directory. Please enter a stage name you’d like to use locally. Or hit enter to use the one based on your AWS credentials (${suggested}): `,
-      (input) => {
-        const final = input.trim() || suggested;
-        State.setStage(paths.appPath, final);
-        rl.close();
-        resolve(final);
-      }
-    );
+    rl.question(question, (input) => {
+      rl.close();
+      resolve(input);
+    });
   });
 }
 

--- a/packages/cli/bin/scripts.mjs
+++ b/packages/cli/bin/scripts.mjs
@@ -271,7 +271,7 @@ async function getStage(argv, config) {
 
   // Re-prompt if stage name is invalid
   while(!State.validateStage(final)) {
-    logger.error(chalk.red("Stage name must start with a letter, and contain only letters, numbers, and hyphens."));
+    logger.error(chalk.red("Stage names must start with a letter, and contain only letters, numbers, and hyphens."));
     const input = await questionSync(`\n${question}`);
     final = input.trim() || suggested;
   }

--- a/packages/core/src/state/index.ts
+++ b/packages/core/src/state/index.ts
@@ -52,3 +52,7 @@ export async function suggestStage() {
     .catch(() => os.userInfo().username);
   return result.replace(/[^A-Za-z0-9]/g, "-");
 }
+
+export function validateStage(stage: string) {
+  return stage.match(/^[A-Za-z][A-Za-z0-9-]*$/);
+}

